log4j-nosql/src/main/java/org/apache/logging/log4j/nosql/appender/mongodb/MongoDbProvider.java

```diff
@@ -97,6 +103,8 @@ public String toString() {
     @PluginFactory
     public static MongoDbProvider createNoSqlProvider(
             @PluginAttribute("collectionName") final String collectionName,
+            @PluginAttribute(value = "capped", defaultBoolean = false) final Boolean isCapped,
```

Comment:
What we've been doing with old factory methods is removing the annotations from them and marking them deprecated. That way it doesn't break custom configurations, but only the new builder is used for normal configuration (although builders are used before factory methods regardless).